internal/proxy/oauthproxy.go

```diff
@@ -1037,6 +1037,7 @@ func (p *OAuthProxy) Authenticate(rw http.ResponseWriter, req *http.Request) (er
 	}
 
 	req.Header.Set("X-Forwarded-User", session.User)
+	req.Header.Set("X-Forwarded-AccessToken", session.AccessToken)
```

Comment:
I don't think we want want to pass the Access Token by default in plaintext in the header to the upstream as it seems like it could escalate to be a security vulnerability. Maybe we can figure out a way to make this a configured opt-in header to set.

Reply:
What do you mean by `Maybe we can figure out a way to make this a configured opt-in header to set`? I have no idea how secure this has to be, I feared an answer like that :).

Reply:
Similar to what is done in [`bitly/oauth2_proxy`](https://github.com/bitly/oauth2_proxy/blob/master/oauthproxy.go#L698) we could conditionally pass this in if the config file has something set like `PassAccessToken`. I think that passing the access token should be done with caution though, as an upstream could potentially do something like log it and create a security vulnerability.

Reply:
Ah nice. Well, I would definitely use such option. If you want me to re-work this PR to offer this feature, let me know ;).

Reply:
That sounds good to me. Thank you for opening and working on this!

Reply:
Also Azure has 4kb access tokens. 😝